tests/integration/targets/infra_join_token/tasks/main.yml

```diff
@@ -0,0 +1,142 @@
+---
+- module_defaults:
+    group/infoblox.bloxone.all:
+      csp_url: "{{ csp_url }}"
+      api_key: "{{ api_key }}"
+
+  block:
+    # Create a random name to avoid conflicts
+    - ansible.builtin.set_fact:
+        name: "test-join-token-{{ 999999 | random | string }}"
+        name2: "test-join-token-{{ 999999 | random | string }}"
+
+    - name: Create a Join Token (check mode)
+      infoblox.bloxone.infra_join_token:
+        name: "{{ name }}"
+        state: present
+      check_mode: true
+      register: join_token
+    - name: Get Information about the Join Token
+      infoblox.bloxone.infra_join_token_info:
+        filters:
+          name: "{{ name }}"
+      register: join_token_info
+    - assert:
+        that:
+          - join_token is changed
+          - join_token_info is not failed
+          - join_token_info.objects | length == 0
+
+    - name: Create a Join Token
+      infoblox.bloxone.infra_join_token:
+        name: "{{ name }}"
+        state: present
+      register: join_token
+    - name: Get Information about the Join Token
+      infoblox.bloxone.infra_join_token_info:
+        filters:
+          name: "{{ name }}"
+      register: join_token_info
+    - assert:
+        that:
+          - join_token is changed
+          - join_token_info is not failed
+          - join_token_info.objects | length == 1
+
+    - name: Create a Join Token (idempotent)
+      infoblox.bloxone.infra_join_token:
+        name: "{{ name }}"
+        state: present
+      register: join_token
+    - assert:
+        that:
+          - join_token is not changed
+          - join_token is not failed
+
+    - name: Delete the Join Token (check mode)
+      infoblox.bloxone.infra_join_token:
+        name: "{{ name }}"
+        state: revoked
+      check_mode: true
+      register: join_token
+    - name: Get Information about the Join Token
+      infoblox.bloxone.infra_join_token_info:
+        filters:
+          name: "{{ name }}"
+      register: join_token_info
+    - assert:
+        that:
+          - join_token is changed
+          - join_token_info is not failed
+          - join_token_info.objects | length == 1
+
+    - name: Delete the Join Token
+      infoblox.bloxone.infra_join_token:
+        name: "{{ name }}"
+        state: revoked
+      register: join_token
+    - name: Get Information about the Join Token
+      infoblox.bloxone.infra_join_token_info:
+        filters:
+          name: "{{ name }}"
+      register: join_token_info
+    - assert:
+        that:
+          - join_token is changed
+          - join_token_info is not failed
+          # The join token is revoked , not deleted , hence the length would be 1
+          - join_token_info.objects | length == 1
```

Comment:
you should also check for the status specifically that it is revoked.

Reply:
Added